src/main/scala/org/apache/mesos/chronos/scheduler/jobs/JobStats.scala

```diff
@@ -80,6 +80,143 @@ class JobStats @Inject()(clusterBuilder: Option[Cluster.Builder], config: Cassan
     jobStates.remove(job.name)
   }
 
+  /**
+   * Queries Cassandra table for past and current job statistics by jobName
+   * and limits by numTasks. The result is not sorted by execution time
+   * @param jobName
+   * @param numTasks
+   * @return list of cassandra rows
+   */
+  private def getJobStatsByName(jobName: String, numTasks: Int): Option[List[Row]] = {
+    var rowsListFinal: Option[List[Row]] = None
+    try {
+      getSession match {
+        case Some(session: Session) =>
+          val query = s"SELECT * FROM ${config.cassandraTable()} WHERE ${JOB_NAME}='${jobName}';"
+          val prepared = statements.getOrElseUpdate(query, {
+            var level = ConsistencyLevel.ANY
+            if (ConsistencyLevel.ANY.name().equalsIgnoreCase(config.cassandraConsistency())) {
```

Comment:
Maybe pull this check out into a separate method like `readConsistencyLevel()`? Then have:

```scala
            session.prepare(
              new SimpleStatement(query)
                .setConsistencyLevel(readConsistencyLevel())
                .asInstanceOf[RegularStatement]
            )
```